maven-shared-utils/src/main/java/org/apache/maven/shared/utils/logging/Style.java

```diff
@@ -81,28 +81,56 @@ void apply( Ansi ansi )
         {
             ansi.a( attribute );
         }
-        if ( color != null )
+
+        if ( fgColor != null )
+        {
+            applyColor( ansi, fgColor );
+        }
+        if ( bgColor != null )
         {
-            ansi.fg( color );
+            applyColor( ansi, bgColor );
         }
     }
 
     @Override
     public String toString()
     {
-        if ( attribute == null && color == null )
+        StringBuilder s = new StringBuilder( name() );
+        if ( attribute != null || fgColor != null || bgColor != null )
         {
-            return name();
+            s.append( "=" );
+            String prefix = "";
+            if ( attribute == null )
+            {
+                prefix = ",";
+                s.append( attribute.toString() );
+            }
+            if ( fgColor == null )
+            {
+                s.append( prefix );
+                prefix = ",";
+                s.append( fgColor );
+            }
+            if ( bgColor == null )
+            {
+                s.append( prefix );
+                s.append( bgColor );
+            }
         }
-        if ( attribute == null )
+        return s.toString();
+    }
+
+    private void applyColor( Ansi ansi, String c )
+    {
+        try
         {
-            return name() + "=" + color.toString();
+            Method m = ansi.getClass().getMethod( c );
+            m.invoke( ansi );
         }
-        if ( color == null )
+        catch ( Exception e )
         {
-            return name() + "=" + attribute.toString();
+            e.printStackTrace();
```

Comment:
If user's configuration doesn't match with an existing method name, then it always fails here.
I think it should fail only once.